omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/NameTemplatesController2_0.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_0;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openmrs.layout.name.NameSupport;
+import org.openmrs.layout.name.NameTemplate;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.WSDoc;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.openmrs.module.webservices.rest.web.v1_0.helper.LayoutTemplateProvider;
+import org.openmrs.serialization.SerializationException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.context.request.WebRequest;
+
+/**
+ * API endpoint to get a list of all available NameTemplates.
+ * <p>
+ * Unfortunately, there already exists a "/nametemplate" endpoint - see NameTemplateController2_0
```

Comment:
Instead of creating a new end point simply to get a list of templates, i would rather that we fix the existing one, even if it means breaking the existing API.  Clients that are not ready to take advantage of this fix can keep using the previous version of the module and upgrade when they are ready to do the necessary changes.

Reply:
OK, I'm happy to do this. However, since that is a bigger change, I will cancel this PR and deliver in two parts:
* first, refactor existing `AddressTemplateController2_0` and `NameTemplateController2_0` to eliminate duplicate code.
* then, make the breaking changes to NameTemplate handling.

I did double check by raising a discussion point in Jira regarding breaking the API, but with no response. On that basis, will proceed.